### PR TITLE
Change Bindable_let_bound's closure_vars to a list

### PR DESCRIPTION
**Pull request by lukemaurer**
This is necessary so that `Let_expr.pattern_match_pair` can operate on
expressions whose closure ids don't match.  The closure ids are no
longer necessary as keys since the right-hand side of the Let_expr is
now kept in order.

### CODE REVIEW

File: middle_end/flambda/naming/bindable_let_bound.ml
Comment (mshinwell) [inline middle_end/flambda/naming/bindable_let_bound.ml]: We've lost the error if the numbers of closure variables are mismatched.  Maybe we could use `List.compare_lengths` to resurrect it.

File: middle_end/flambda/simplify/simplify_expr.rec.ml
Comment (mshinwell) [inline middle_end/flambda/simplify/simplify_expr.rec.ml]: Double space before `in`

**Review by mshinwell — APPROVED** [review]
This is ok to merge after addressing a couple of minor comments.  Please make sure the CI is green before merging.

File: middle_end/flambda/simplify/simplify_expr.rec.ml
Comment (lukemaurer) [reply middle_end/flambda/simplify/simplify_expr.rec.ml]: Fixed.

File: middle_end/flambda/naming/bindable_let_bound.ml
Comment (lukemaurer) [reply middle_end/flambda/naming/bindable_let_bound.ml]: Good idea; done.

**Pull request merged**
